lib/wherebot.rb

```diff
@@ -46,16 +46,21 @@ def initialize(imap, email_id, destroy = false)
     end
 
     def create
-      WhereMessage.find_or_create_by(
+      wm = WhereMessage.find_or_create_by(
         provenance: 'where@eff.org',
         sent_at: DateTime.parse(headers.date),
-        body: body,
         user: User.find_by(
           email: "#{headers.from[0].mailbox}@#{headers.from[0].host}"
         )
       )
+      if wm.update(body: body)
+        destroy_message
+      else
+        storage = ENV['where_failures'].try(:dup) || ''
+        storage << wm.to_json
+        ENV['where_failures'] = storage
```

Comment:
Writing JSON to an environment variable seems like an unusual way to record errors (if I'm understanding this code correctly). Any particular reason you chose that approach?

Reply:
You are right, it is unusual!  I did it because it seemed fast, and then wrote myself a note to come back to it.  My second choice was to write them to a logfile, and Peter suggested that we set FAB up with Sentry and allow them to surface.